engine/src/scriptpt.cpp

```diff
@@ -1310,33 +1309,80 @@ Parse_stat MCScriptPoint::lookup(Script_point t, const LT *&dlt)
 	return PS_NO_MATCH;
 }
 
-bool MCScriptPoint::lookupconstantvalue(const char *& r_value)
+bool MCScriptPoint::lookupconstantintable(int& r_position)
 {
-	uint2 high = constant_table_size;
-	uint2 low = 0;
-	int4 cond;
+    int high = constant_table_size;
+    int low = 0;
+    int cond;
     
     MCAutoStringRefAsCString t_token;
     t_token . Lock(gettoken_stringref());
     const char *token_cstring = *t_token;
-	while (low < high)
-	{
-		uint2 mid = low + ((high - low) >> 1);
-		cond = MCU_strncasecmp(token_cstring, constant_table[mid].token, token.getlength());
-		if (cond == 0)
-			cond -= constant_table[mid].token[token.getlength()];
-		if (cond < 0)
-			high = mid;
-		else
-			if (cond > 0)
-				low = mid + 1;
-			else
-			{
-                r_value = constant_table[mid] . svalue;
-				return true;
-			}
-	}
-	return false;
+    while (low < high)
+    {
+        int mid = low + ((high - low) >> 1);
+        cond = MCU_strncasecmp(token_cstring, constant_table[mid].token, token.getlength());
+        if (cond == 0)
+        {
+            cond -= constant_table[mid].token[token.getlength()];
+        }
+        
+        if (cond < 0)
+        {
+            high = mid;
+        }
+        else
+        {
+            if (cond > 0)
+            {
+                low = mid + 1;
+            }
+            else
+            {
+                r_position = mid;
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool MCScriptPoint::constantnameconvertstoconstantvalue()
+{
+    int t_position;
+    if (!lookupconstantintable(t_position))
+        return false;
+    
+    switch (constant_table[t_position].type)
+    {
+        case kCValueTypeString:
+        {
+            MCNewAutoNameRef t_constant_value;
+            MCNameCreateWithNativeChars((const char_t *)constant_table[t_position].string,
+                                        strlen(constant_table[t_position].string),
+                                        &t_constant_value);
+
+            return MCNameIsEqualTo(gettoken_nameref(), *t_constant_value, kMCCompareExact);
```

Comment:
There's no need to convert to a name her e- just use MCStringIsEqualToCString with gettoken_stringref() and the string from the table.